data-prepper-plugins/opensearch-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/NoSearchContextWorkerTest.java

```diff
@@ -77,16 +84,35 @@ public class NoSearchContextWorkerTest {
     @Mock
     private AcknowledgementSetManager acknowledgementSetManager;
 
+    @Mock
+    private OpenSearchSourcePluginMetrics openSearchSourcePluginMetrics;
+
+    @Mock
+    private Counter documentsProcessedCounter;
+
+    @Mock
+    private Counter indicesProcessedCounter;
+
+    @Mock
+    private Counter processingErrorsCounter;
+
+    @Mock
+    private Timer indexProcessingTimeTimer;
+
     private ExecutorService executorService;
 
     @BeforeEach
     void setup() {
         executorService = Executors.newSingleThreadExecutor();
         lenient().when(openSearchSourceConfiguration.isAcknowledgmentsEnabled()).thenReturn(false);
+        lenient().when(openSearchSourcePluginMetrics.getDocumentsProcessedCounter()).thenReturn(documentsProcessedCounter);
```

Comment:
It might be a good idea to make the mocks themselves lenient. Then you can simplify this part.

```
@Mock(lenient = true)
```